src/Nethermind/Nethermind.Api/Extensions/IPlugin.cs

```diff
@@ -33,5 +33,9 @@ public interface INethermindPlugin : IAsyncDisposable
         Task InitNetworkProtocol();
         
         Task InitRpcModules();
+
+        Task BeforeStep(string name) => Task.CompletedTask;
+
+        Task AfterStep(string name) => Task.CompletedTask;
```

Comment:
In .net event are named with `On` prefix like OnBeforeStep, OnAfterStep.

Also as there is a good chance this code  won't actually be async would change to ValueTask.

Reply:
Would change parameter to be StepInfo, just make Stage set there internal.